modules/data-entry/src/main/frontend/src/questionnaire/SelectableAreaQuestion.jsx

```diff
@@ -0,0 +1,252 @@
+      for (let i = 0; i < selections.length; i++) {
+        if (selections[i][VALUE_POS] === mapEntry.value) {
+          // Area has been selected, highlight it.
+          mapEntry.preFillColor = highlightColor ? highlightColor : mapEntry.fillColor;
+          found = true;
+          break;
+        }
+      }
+      if (!found) {
+        // Area has not been selected, clear any highlight.
+        delete mapEntry.preFillColor;
+      }
+    }))
+  }
+
+  // As the displayed values are based on an extension, the backend doesn't know what display values go with what values.
+  // Once the map has loaded, initialize the selections to set the approriate display values.
+  useEffect(() => {
+    if(!initialized && map) {
+      selections.forEach(selection =>{
+        map.forEach(mapEntry => {
+          if (mapEntry.value === selection[LABEL_POS] && mapEntry.title) {
+            selection[LABEL_POS] = mapEntry.title;
+          }
+        });
+      })
+
+      // Update both the map highlights and the displayed list of selections
+      updateMapWithSelections();
+      updateSelectionsDisplay();
+      setInitialized(true);
+    }
+  }, [initialized, map])
+
+  // List out the selected areas in text
+  let updateSelectionsDisplay = () => {
+    setSelectionDisplay(
+      <ul>
+        {selections.map(selection => {
+          return <li key={selection[VALUE_POS]}>{selection[LABEL_POS]}</li>
+        })}
+      </ul>);
+  }
+
+  // When the question's variant changes, load the relevant data.
+  useEffect(() => {
+    if (variant) {
+      let unparsedMap = variant.map?.["jcr:content"]?.["jcr:data"];
+      setMap(unparsedMap ? JSON.parse(unparsedMap) : null);
+      setImageUrl(variant.image);
+      setMaxWidth(variant.maxWidth);
+      setStrokeColor(variant.strokeColor);
+      setHighlightColor(variant.highlightColor);
+    } else {
+      resetQuestionData();
+    }
+  }, [variant])
+
+  // Clear out any data associated with a specific question
+  let resetQuestionData = () => {
+    setImageUrl(null);
+    setMap(null);
+    setSelections(initialSelection);
+    setMaxWidth(null);
+  }
+
+  // When an area is clicked, update the selection associated with that area.
+  let onAreaClicked = (area, index) => {
+    let clickedEntry = [area.title, area.value];
+
+    if (maxAnswers == 1) {
+      // Single select: Just set selection to the most recent value
+      selections.pop();
+      selections.push(clickedEntry);
+    } else {
+      // Multi select: toggle the values' selection state
+      let entryIndex = -1;
+      selections.forEach((selectionValue, selectionIndex) => {
+        if (selectionValue[VALUE_POS] === clickedEntry[VALUE_POS]) {
+          entryIndex = selectionIndex;
+        }
+      })
+      if(entryIndex == -1) {
+        selections.push(clickedEntry);
+      } else {
+        selections.splice(entryIndex, 1);
+      }
+    }
+
+    let newSelections = selections.concat()
+    setSelections(newSelections)
```

Comment:
This might end up buggy and unpredictable.

It's recommended to do all the processing within the state setter, something like this (not tested):

```suggestion
    setSelections(oldSel => {
      if (maxAnswers == 1) {
        // Single select: Just set selection to the most recent value
        return [clickedEntry];
      }
      // Multi select: toggle the values' selection state
      let newSel = oldSel.slice();
      let entryIndex = -1;
      newSel.forEach((selectionValue, selectionIndex) => {
        if (selectionValue[VALUE_POS] === clickedEntry[VALUE_POS]) {
          entryIndex = selectionIndex;
        }
      })
      if(entryIndex == -1) {
        newSel.push(clickedEntry);
      } else {
        oldSel.splice(entryIndex, 1);
      }
      return newSel;
    });
```